vita3k/cpu/src/dynarmic_cpu.cpp

```diff
@@ -127,6 +128,13 @@ class ArmDynarmicCallback : public Dynarmic::A32::UserCallbacks {
         Ptr<T> ptr{ addr };
         if (!ptr || !ptr.valid(*parent->mem)) {
             LOG_WARN("Invalid read of uint{}_t at address: 0x{:x}", sizeof(T) * 8, addr);
+            LOG_WARN("{}", this->cpu->save_context().description());
+
+            auto pc = this->cpu->get_pc();
+            if (pc == 0)
+                LOG_CRITICAL("PC is 0");
```

Comment:
null was just an example. In emu first memory page is always bad ( this->cpu->get_pc() < parent->mem->page_size), and for any other value I don't know, and I don't know how to check is pointer bad or not.

Reply:
ic, afaik the only way to check if the pointer is actually valid when the address looks right is to actually check its value, which would cause a segfault, however it is still better to have some checks than nothing like it was before

Reply:
First page is always bad. See https://github.com/Vita3K/Vita3K/blob/054a8119096596c65007e54bf3db49dc69d450b3/vita3k/mem/src/mem.cpp#L103-L109
My implementation of dynarmic error logging. maybe it explain my idea a bit better https://github.com/Vita3K/Vita3K/commit/5d86be362ffcd9b308a580066408a738343a7465